src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/SubscriptionReceiver.java

```diff
@@ -0,0 +1,39 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Networking component that handles receiving subscriptions.
+ */
+public class SubscriptionReceiver {
+
+  private static final Logger LOG = LogManager.getLogger(SubscriptionReceiver.class);
+  private final NetworkRequestQueue<CompositeSubscribeRequest> rxQ;
+  private final ScheduledExecutorService threadPool;
+  private final SubscriptionReceiverTask subscriptionReceiverTask;
+
+  public SubscriptionReceiver(
+      NetworkRequestQueue<CompositeSubscribeRequest> rxQ,
+      ScheduledExecutorService threadPool,
+      SubscriptionReceiverTask subscriptionReceiverTask) {
+    this.rxQ = rxQ;
+    this.threadPool = threadPool;
+    this.subscriptionReceiverTask = subscriptionReceiverTask;
+  }
+
+  public boolean enqueue(final CompositeSubscribeRequest compositeSubscribeRequest) {
+    return rxQ.offer(compositeSubscribeRequest);
+  }
+
+  public void start() {
+    threadPool.scheduleAtFixedRate(subscriptionReceiverTask, 0, 250, TimeUnit.MILLISECONDS);
```

Comment:
How did we arrive at this number ? Can we move this to RcaConst class ?

Reply:
The scheduler's periodicity is 1s, so if we wanted to receive and send in that 1s, setting send and receive periodicity to 250ms will give us two chances(but I guess that explanation doesn't hold water because there is no synchronization across schedulers, now that I think about it), but the higher the frequency, the more chances you get to send/receive - was my thinking. I've changed it from being a poll based model to push based where we queue the message up by calling execute on a threadpoolexecutor as soon as we can so it's not needed anymore.